packages/runtime-common/realm.ts

```diff
@@ -594,6 +594,13 @@ export class Realm {
         return Array.isArray(sourceValue) ? sourceValue : undefined;
       }
     );
+
+    if (patch.data.relationships) {
+      // relationships should be overwritten instead of merged, otherwise we
+      // won't be able to remove items from linksToMany relationships
+      card.data.relationships = patch.data.relationships;
+    }
```

Comment:
Does the JSON-API spec have an opinion about this? This seems a little contrary to the idea of patch. I'd expect a null-value-aware merge would be appropriate here?

Reply:
i think burcu got this idea from how we handle the contains-many relationships. but yeah i agree, this does seem antithetical to to how patch should work

Reply:
yes, I got the idea from how we're handling the containsMany arrays above this code. I'll see how else we can handle these but I'm not sure... If there are 2 items and the first one is removed, the second item needs to become the first and only item. We don't want first item to show `null` and second item to display a link. If all items are removed, then we need to change the link to `null`. 

linksToMany with items:
```
relationships: {
            'pets.0': {
              links: {
                self: `${testRealmURL}dir/mango`,
              },
            },
            'pets.1': {
              links: {
                self: `${testRealmURL}dir/van-gogh`,
              },
            },
          },
```

linksToMany with no items:

```
relationships: {
            'pets': {
              links: {
                self: null,
              },
            },
          },
```

Reply:
Of course the previous solution was simpler, but it required passing in the full relationships object, instead of just the changed part. The current version allows for just passing in the change in the request (see the two tests I added in `realm-test`).